### PR TITLE
Add frontend user login and logout

### PR DESCRIPTION
**Pull request by jasalisbury**
**Description of change**

A user that has not logged in will be shown a 'landing' page with a login button presented. Once logged in the user will view a home page with a logout button, and the header will populate with links (home and the admin page are the only links right now). When the logout button is pressed a logout message is displayed.

The API will now return a 401 when accessed without a valid logged in session. Hitting `http://localhost:8080/api/login` (`http://localhost:3000/api/login` when proxied) will trigger the login process. 

**How to test**

 1) Populate `LOGIN_REDIRECT_URL` in the server's `.env` file (example and value you should use is in `.env.example`)
 2) `yarn deps` (`yarn docker:deps`)
 3) `yarn start` (`yarn docker:start`)
 4) Browse to `http://localhost:3000`. You should see a page prompting you to login with no navigation links
 5) Login to HSES via the button on step 3
 6) After login you should be back at our frontend and see "Welcome to the TTA Smart Hub" followed by your email address
 7) Click "logout" and you will be returned to the page on step 3

**Notes**

 * ~@SarahJaine we will have a new env variable to set for the backend. It is not a secret. It's name is `LOGIN_REDIRECT_URL` and is the url the server should redirect to after login to HSES (so the url of the frontend). I can poke around and figure out where to add it just a heads up that I might ping you if I can't figure it out.~ Figured it out, see comment below
 * ~An env variable was added to the frontend (named `REACT_APP_API_URL`) as well that is not a secret (we can't have frontend secrets anyways because they will always be sent along with the frontend HTML/javascript in plaintext). It will need to be set to the backend API URL in each environment at build time. @SarahJaine I might ping you again if I can't figure it out.~
 * API routes are now prefixed with `/api`. The OAuth callback hasn't been updated (we can't directly do the update) so it looks a bit goofy still having a non-prefixed route. It was required to keep login working. If we decide we want to prefix with `/api` and the callback has been updated in HSES it will be an easy switch for the backend. 
 * I imagine there will be updates to the user endpoint and calling code in the frontend once we create the users/permissions database tables

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/4

**Checklist**

- [x] Meets issue criteria
- [x] Code tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

**Comment by jasalisbury** [comment]
You'll notice duplicated env varibles, `TTA_SMART_HUB_URI` and `REDIRECT_URI_HOST`. I propose we move to using `TTA_SMART_HUB_URI` and remove `REDIRECT_URI_HOST`. `REDIRECT_URI_HOST` cannot be removed in this PR because of the localhost OAuth callback is set to a value that differs from the value `TTA_SMART_HUB_URI` has (`localhost:3000` vs `localhost:8080`). Once the OAuth localhost callback is updated we would be able to remove the `REDIRECT_URI_HOST` environment variable.

**Comment by rahearn** [comment]
> Once the OAuth localhost callback is updated we would be able to remove the `REDIRECT_URI_HOST` environment variable.

Just a quick note, the oauth redirect has to stay at `localhost:8080` to work with the HSES dev token we have, but I'm fine with getting rid of the `3000` port and moving everything onto `8080` if you wish.

File: frontend/src/fetchers/Auth.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import join from 'url-join';
+
+const callApi = async (url) => {
+  const res = await fetch(url, {
```
Comment (kryswisnaskas) [inline frontend/src/fetchers/Auth.js:4]: Have you thought of using `axios` here instead to be consistent between the backend and frontend?

**Comment by kryswisnaskas** [comment]
Overall it looks really good! The UI is great (even though it hasn't been seen by the design yet). 

I am ok with adding `/api` to the routes. 👍  We need to also add a way to version the backend apis. The straightforward way would be with adding a version number like `v1` to the path.

Even though related to a different PR, could we remove the `Admin` link from the UI here, but keep the functionality via a direct url? This way we will eliminate the impression that the `Admin` link is going to be a part of the finished product.

File: frontend/src/__tests__/App.js [hunk 2]
```diff
@@ -1,9 +1,44 @@
...
+    });
+  });
+
+  describe('when unauthenticated', () => {
+    beforeEach(() => {
+      fetchMock.get(userUrl, 401);
+      render(<App />);
+    });
+
+    it('displays the logout button', async () => {
```
Comment (rahearn) [inline frontend/src/__tests__/App.js:40]: Future cleanup: this should be `displays the login button`

File: frontend/src/fetchers/Auth.js (hunk 1 above)
Comment (rahearn) [reply frontend/src/fetchers/Auth.js:4]: We should support IE 11, so either `axios` or a shim for `fetch` is needed. If a shim isn't already in place I'm fine with adding that to the list of things to change after the fact.

File: src/index.test.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ import server from './index';
 describe('Root', () => {
   test('Redirects to login if user is not logged in', async () => {
```
Comment (rahearn) [inline src/index.test.js:5]: Future cleanup: This test name should be updated to reflect current behavior

**Review by rahearn — APPROVED** [review]
See my inline comments for some cleanup tasks I'd like to see added to the next stories to be worked on, but I'm fine with merging this as-is. 🎉

**Comment by kryswisnaskas** [comment]
Merging this to main to allow merge to the HHS upstream for the demo, while any cleanup work can be addressed in a separate PR.

File: frontend/src/fetchers/Auth.js (hunk 1 above)
Comment (jasalisbury) [reply frontend/src/fetchers/Auth.js:4]: This work will be done here https://github.com/HHS/Head-Start-TTADP/issues/71

**Pull request merged**